src/server/coprocessor/executor/scanner.rs

```diff
@@ -0,0 +1,385 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use server::coprocessor::Result;
+use kvproto::coprocessor::KeyRange;
+use storage::{Key, Value, SnapshotStore, Statistics, ScanMode};
+use util::escape;
+
+// Scanner for TableScan and IndexScan
```

Comment:
Make this comment more descriptive like:
`Scanner` is a helper struct to wrap all common codes for `TableScanExecutor` and `IndexScanExecutor`.